Core/OfficeDevPnP.Core/AuthenticationManager.cs

```diff
@@ -520,6 +520,105 @@ public ClientContext GetHighTrustCertificateAppOnlyAuthenticatedContext(string s
 
             return clientContext;
         }
+
+        /// <summary>
+        /// Returns a SharePoint ClientContext using High Trust Certificate App Only Authentication
+        /// </summary>
+        /// <param name="siteUrl">Site for which the ClientContext object will be instantiated</param>
+        /// <param name="clientId">The SharePoint Client ID</param>
+        /// <param name="certificatePath">Full path to the private key certificate (.pfx) used to authenticate</param>
+        /// <param name="certificatePassword">Password used for the private key certificate (.pfx)</param>
+        /// <param name="certificateIssuerId">The IssuerID under which the CER counterpart of the PFX has been registered in SharePoint as a Trusted Security Token issuer</param>
+        /// <param name="loginName">
+        /// Name of the user (login name) on whose behalf to create the access token.
+        /// Supported input formats are SID and User Principal Name (UPN).
+        /// If the parameter is left empty (including null) an App Only Context will be created.
+        /// </param>
+        /// <returns>Authenticated SharePoint ClientContext</returns>
+        public ClientContext GetHighTrustCertificateAppAuthenticatedContext(string siteUrl, string clientId, string certificatePath, string certificatePassword, string certificateIssuerId, string loginName)
+        {
+            var certPassword = Utilities.EncryptionUtility.ToSecureString(certificatePassword);
+            return GetHighTrustCertificateAppAuthenticatedContext(siteUrl, clientId, certificatePath, certPassword, certificateIssuerId, loginName);
```

Comment:
@tmeckel 
There's a little compile error.
You are using a not defined variable named "certPassword" (Line 541 ) but the parameter name of the method is "certificatePassword" (line 538).